File: com.bosch.fsp.master_of_plow.gen/src/com/bosch/nevonex/fcal/impl/ACEINNA_ANGLES_T_R.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.fcal.impl;

import com.bosch.nevonex.fcal.IACEINNA_ANGLES_T_R;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.EObject;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.impl.EObjectImpl;

/**
 * @generated
 */
public class ACEINNA_ANGLES_T_R extends EObjectImpl implements IACEINNA_ANGLES_T_R {
	protected float pitch = 0.0f;
	protected float roll = 0.0f;
	protected int pitch_Compensation = 0;
	protected int pitch_FigureOfMerit = 0;
	protected int roll_Compensation = 0;
	protected int roll_FigureOfMerit = 0;
	protected float pitchRoll_Latency = 0.0f;

	protected ACEINNA_ANGLES_T_R() {
		super();
	}

	@Override
	protected EClass eStaticClass() {
		return FcalPackage.Literals.ACEINNA_ANGLES_T_R;
	}

	public float getPitch() {
		return pitch;
	}

	public void setPitch(float newPitch) {
		pitch = newPitch;
	}

	public float getRoll() {
		return roll;
	}

	public void setRoll(float newRoll) {
		roll = newRoll;
	}

	public int getPitch_Compensation() {
		return pitch_Compensation;
	}

	public void setPitch_Compensation(int newPitch_Compensation) {
		pitch_Compensation = newPitch_Compensation;
	}

	public int getPitch_FigureOfMerit() {
		return pitch_FigureOfMerit;
	}

	public void setPitch_FigureOfMerit(int newPitch_FigureOfMerit) {
		pitch_FigureOfMerit = newPitch_FigureOfMerit;
	}

	public int getRoll_Compensation() {
		return roll_Compensation;
	}

	public void setRoll_Compensation(int newRoll_Compensation) {
		roll_Compensation = newRoll_Compensation;
	}

	public int getRoll_FigureOfMerit() {
		return roll_FigureOfMerit;
	}

	public void setRoll_FigureOfMerit(int newRoll_FigureOfMerit) {
		roll_FigureOfMerit = newRoll_FigureOfMerit;
	}

	public float getPitchRoll_Latency() {
		return pitchRoll_Latency;
	}

	public void setPitchRoll_Latency(float newPitchRoll_Latency) {
		pitchRoll_Latency = newPitchRoll_Latency;
	}

	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH:
			return getPitch();
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL:
			return getRoll();
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_COMPENSATION:
			return getPitch_Compensation();
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_FIGURE_OF_MERIT:
			return getPitch_FigureOfMerit();
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL_COMPENSATION:
			return getRoll_Compensation();
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL_FIGURE_OF_MERIT:
			return getRoll_FigureOfMerit();
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_ROLL_LATENCY:
			return getPitchRoll_Latency();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH:
			setPitch((Float) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL:
			setRoll((Float) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_COMPENSATION:
			setPitch_Compensation((Integer) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_FIGURE_OF_MERIT:
			setPitch_FigureOfMerit((Integer) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL_COMPENSATION:
			setRoll_Compensation((Integer) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL_FIGURE_OF_MERIT:
			setRoll_FigureOfMerit((Integer) newValue);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_ROLL_LATENCY:
			setPitchRoll_Latency((Float) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH:
			setPitch(0.0f);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL:
			setRoll(0.0f);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_COMPENSATION:
			setPitch_Compensation(0);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_FIGURE_OF_MERIT:
			setPitch_FigureOfMerit(0);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL_COMPENSATION:
			setRoll_Compensation(0);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL_FIGURE_OF_MERIT:
			setRoll_FigureOfMerit(0);
			return;
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_ROLL_LATENCY:
			setPitchRoll_Latency(0.0f);
			return;
		}
		super.eUnset(featureID);
	}

	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH:
			return pitch != 0.0f;
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL:
			return roll != 0.0f;
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_COMPENSATION:
			return pitch_Compensation != 0;
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_FIGURE_OF_MERIT:
			return pitch_FigureOfMerit != 0;
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL_COMPENSATION:
			return roll_Compensation != 0;
		case FcalPackage.ACEINNA_ANGLES_T_R__ROLL_FIGURE_OF_MERIT:
			return roll_FigureOfMerit != 0;
		case FcalPackage.ACEINNA_ANGLES_T_R__PITCH_ROLL_LATENCY:
			return pitchRoll_Latency != 0.0f;
		}
		return super.eIsSet(featureID);
	}
} //ACEINNA_ANGLES_T_R

File: master_of_plow_CPP_SDK/src-gen/nevonex/fcal/ACEINNA_ANGLES_T_R.cpp
/*
 * nevonex/fcal/ACEINNA_ANGLES_T_R.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "ACEINNA_ANGLES_T_R.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/fcal/FcalPackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <algorithm>
#include <nevonex/fcb/SubscriberEnum.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

#if BOOST_VERSION >= 106501
#define BOOST_STACKTRACE_HEADERS_FOUND
#include <boost/stacktrace.hpp>
#endif

/*PROTECTED REGION ID(ACEINNA_ANGLES_T_R.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::fcal;
using namespace ::nevonex::log;

// Default constructor
ACEINNA_ANGLES_T_R::ACEINNA_ANGLES_T_R() : m_pitch(0.0f), m_roll(0.0f), m_pitch_Compensation(0), m_pitch_FigureOfMerit(0), m_roll_Compensation(0), m_roll_FigureOfMerit(0), m_pitchRoll_Latency(0.0f)
{

    /*PROTECTED REGION ID(ACEINNA_ANGLES_T_R__ACEINNA_ANGLES_T_R) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

ACEINNA_ANGLES_T_R::~ACEINNA_ANGLES_T_R()
{

    /*PROTECTED REGION ID(ACEINNA_ANGLES_T_R__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

// Attributes

::ecore::EFloat ACEINNA_ANGLES_T_R::getPitch() const
{
    return m_pitch;
}

void ACEINNA_ANGLES_T_R::setPitch(::ecore::EFloat _pitch)
{

    ::ecore::EFloat _old_pitch = m_pitch;
    m_pitch = _pitch;

}

::ecore::EFloat ACEINNA_ANGLES_T_R::getRoll() const
{
    return m_roll;
}

void ACEINNA_ANGLES_T_R::setRoll(::ecore::EFloat _roll)
{

    ::ecore::EFloat _old_roll = m_roll;
    m_roll = _roll;

}

::ecore::EInt ACEINNA_ANGLES_T_R::getPitch_Compensation() const
{
    return m_pitch_Compensation;
}

void ACEINNA_ANGLES_T_R::setPitch_Compensation(::ecore::EInt _pitch_Compensation)
{

    ::ecore::EInt _old_pitch_Compensation = m_pitch_Compensation;
    m_pitch_Compensation = _pitch_Compensation;

}

::ecore::EInt ACEINNA_ANGLES_T_R::getPitch_FigureOfMerit() const
{
    return m_pitch_FigureOfMerit;
}

void ACEINNA_ANGLES_T_R::setPitch_FigureOfMerit(::ecore::EInt _pitch_FigureOfMerit)
{

    ::ecore::EInt _old_pitch_FigureOfMerit = m_pitch_FigureOfMerit;
    m_pitch_FigureOfMerit = _pitch_FigureOfMerit;

}

::ecore::EInt ACEINNA_ANGLES_T_R::getRoll_Compensation() const
{
    return m_roll_Compensation;
}

void ACEINNA_ANGLES_T_R::setRoll_Compensation(::ecore::EInt _roll_Compensation)
{

    ::ecore::EInt _old_roll_Compensation = m_roll_Compensation;
    m_roll_Compensation = _roll_Compensation;

}

::ecore::EInt ACEINNA_ANGLES_T_R::getRoll_FigureOfMerit() const
{
    return m_roll_FigureOfMerit;
}

void ACEINNA_ANGLES_T_R::setRoll_FigureOfMerit(::ecore::EInt _roll_FigureOfMerit)
{

    ::ecore::EInt _old_roll_FigureOfMerit = m_roll_FigureOfMerit;
    m_roll_FigureOfMerit = _roll_FigureOfMerit;

}

::ecore::EFloat ACEINNA_ANGLES_T_R::getPitchRoll_Latency() const
{
    return m_pitchRoll_Latency;
}

void ACEINNA_ANGLES_T_R::setPitchRoll_Latency(::ecore::EFloat _pitchRoll_Latency)
{

    ::ecore::EFloat _old_pitchRoll_Latency = m_pitchRoll_Latency;
    m_pitchRoll_Latency = _pitchRoll_Latency;

}

// References

